lib/components/AssociatedGenesTable.js

```diff
@@ -0,0 +1,173 @@
+    .find(schema => schema.sourceId === sourceId)
+    .tissues.forEach(tissue => {
+      columns.push({
+        id: tissue.id,
+        label: tissue.name,
+        renderCell: rowData => {
+          return rowData[tissue.id]
+            ? rowData[tissue.id].toPrecision(3)
+            : 'No information';
+        },
+      });
+    });
+  return columns;
+};
+
+const getTissueData = (genesForVariantSchema, genesForVariant, sourceId) => {
+  const data = [];
+  let searchField;
+
+  genesForVariantSchema.qtls.forEach(qtl => {
+    if (qtl.sourceId === sourceId) {
+      searchField = 'qtls';
+    }
+  });
+
+  !searchField &&
+    genesForVariantSchema.intervals.forEach(interval => {
+      if (interval.sourceId === sourceId) {
+        searchField = 'intervals';
+      }
+    });
+
+  !searchField &&
+    genesForVariantSchema.functionalPredictions.forEach(fp => {
+      if (fp.sourceId === sourceId) {
+        searchField = 'functionalPredictions';
+      }
+    });
+
+  genesForVariant.forEach(geneForVariant => {
+    const row = { gene: geneForVariant.gene.symbol };
+    const element = geneForVariant[searchField].find(
+      item => item.sourceId === sourceId
+    );
+
+    if (element) {
+      element.tissues.forEach(elementTissue => {
+        row[elementTissue.tissue.id] =
+          elementTissue.score || elementTissue.beta;
+      });
+    }
+
+    data.push(row);
+  });
+  return data;
+};
+
+class AssociatedGenesTable extends Component {
+  state = {
+    value: 0,
+  };
+
+  handleChange = (_, value) => {
+    this.setState({ value });
+  };
+
+  render() {
+    const { value } = this.state;
+    const { genesForVariantSchema, genesForVariant } = this.props.data;
+
+    // Hardcoding the order and assuming qtls, intervals, and
+    // functionalPredictions are the only fields in the schema
+    const schemas = [
+      ...genesForVariantSchema.qtls,
+      ...genesForVariantSchema.intervals,
+      ...genesForVariantSchema.functionalPredictions,
+    ];
+
+    const filteredSchemas = schemas.filter(schema => schema.tissues.length > 1);
+
+    const columnsAll = getColumnsAll(schemas);
+    const dataAll = getDataAll(genesForVariant);
+
+    return (
+      <Fragment>
+        <Tabs value={value} onChange={this.handleChange}>
+          <Tab label="All" />
+          {filteredSchemas.map(schema => {
+            return (
+              <Tab
+                key={schema.sourceId}
+                value={schema.sourceId}
+                label={schema.sourceId}
+              />
+            );
+          })}
+        </Tabs>
+        {value === 0 && <OtTable columns={columnsAll} data={dataAll} />}
+        {filteredSchemas.map(schema => {
+          const tissueColumns = getTissueColumns(schemas, schema.sourceId);
+          const tissueData = getTissueData(
+            genesForVariantSchema,
+            genesForVariant,
+            schema.sourceId
+          );
+          return (
+            value === schema.sourceId && (
```

Comment:
A minor point, but here you calculate `tissueColumns` and `tissueData` regardless of whether `value === schema.sourceId`. If you refactored to the following, the calculation would only occur if that tab was visible.
```
        {filteredSchemas.map(schema => {
          return (
            value === schema.sourceId && (
              <OtTable
                key={schema.sourceId}
                columns={getTissueColumns(schemas, schema.sourceId)}
                data={getTissueData(
                  genesForVariantSchema,
                  genesForVariant,
                  schema.sourceId
                )}
              />
            )
          );
        })}
```